app/controllers/admin/api_controller.rb

```diff
@@ -0,0 +1,54 @@
+class Admin::ApiController < ApplicationController
+  before_filter :admin_only
+  before_filter :check_for_cancel, only: [:create, :update]
+
+  def index
```

Comment:
Sorry but can we store this is redis and have a task which migrates it to the database periodically. See how we are recording ip addresses....